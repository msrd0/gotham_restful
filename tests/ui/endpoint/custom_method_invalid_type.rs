#[macro_use]
extern crate gotham_restful;

#[derive(Resource)]
#[resource(read_all)]
struct FooResource;

#[endpoint(method = "String::new()", uri = "custom_read")]
async fn read_all() {}

fn main() {}
