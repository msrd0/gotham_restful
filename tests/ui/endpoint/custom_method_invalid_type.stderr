error[E0308]: mismatched types
 --> $DIR/custom_method_invalid_type.rs:8:21
  |
8 | #[endpoint(method = "String::new()", uri = "custom_read")]
  | --------------------^^^^^^^^^^^^^^^-----------------------
  | |                   |
  | |                   expected struct `Method`, found struct `std::string::String`
  | expected `Method` because of return type
